src/edtr-io/plugins/image.tsx

```diff
@@ -85,44 +107,70 @@ export function createUploadImageHandler(getCsrfToken: () => string) {
   }
 }
 
-export function createReadFile(getCsrfToken: () => string) {
+export function createReadFile() {
   return function readFile(file: File): Promise<LoadedFile> {
     return new Promise((resolve, reject) => {
-      const reader = new FileReader()
-
-      reader.onload = function (e: ProgressEvent) {
-        if (!e.target) return
-        const formData = new FormData()
-        formData.append('attachment[file]', file)
-        formData.append('type', 'file')
-        formData.append('csrf', getCsrfToken())
-
-        fetch('/attachment/upload', {
-          method: 'POST',
-          body: formData,
-        })
-          .then((response) => response.json())
-          .then((data: { success: boolean; files: { location: string }[] }) => {
-            if (!data['success']) reject()
-            resolve({
-              file,
-              dataUrl: data.files[0].location,
-            })
-          })
-          .catch(() => {
-            reject()
+      // hard to get to the default auth logic outside of react components
+      // I provide dummy functions for refresh / clear token here since we only need the token
+      // and can be quite certain, that the token has been checked when loading the edtr in the first place
+      const authenticationPayload = parseAuthCookie(
+        (_dummy: string) => new Promise(() => {}),
+        () => {}
+      )
+      if (!authenticationPayload?.token) return
+
+      const gqlFetch = createAuthAwareGraphqlFetch({
+        current: authenticationPayload,
+      })
+      const args = JSON.stringify({
+        query: uploadUrlQuery,
+        variables: {
+          mediaType: mimeTypesToMediaType[file.type as SupportedMimeType],
```

Comment:
I have the feeling that we should check `file.type` instead of the file name extension -> that is more save...